rpc/dial.go

```diff
@@ -294,9 +295,9 @@ func dial(
 	return conn, cached, nil
 }
 
-func lookupMDNSCandidate(ctx context.Context, address string, logger golog.Logger) (*zeroconf.ServiceEntry, error) {
+func lookupMDNSCandidate(ctx context.Context, address string, logger utils.ZapCompatibleLogger) (*zeroconf.ServiceEntry, error) {
 	candidates := []string{address, strings.ReplaceAll(address, ".", "-")}
-	resolver, err := zeroconf.NewResolver(logger, zeroconf.SelectIPRecordType(zeroconf.IPv4))
+	resolver, err := zeroconf.NewResolver(logger.Desugar().Sugar(), zeroconf.SelectIPRecordType(zeroconf.IPv4))
```

Comment:
Calling out this `Desugar` + `Sugar` pattern here for discussion. Sounds like we need this to go from an RDK `impl` logger to zap specifically when passing to `zeroconf` API.

The zeroconf library is our fork (well, Eric's, strictly speaking), and we actually added these `logger` arguments at some point as the main fork of `zeroconf` used the standard library's log method and printed a bunch of stuff uncontrollably to STDERR.

There are two options in my head. We can:
- `Desugar` and `Sugar` when passing to zeroconf
- Refactor `zeroconf` to accept either a copy of `utils.ZapCompatibleLogger` or directly reference it (the latter would create an annoying circular dependency)

I think the first option (what you have) is fine, but happy to hear arguments to the contrary.

Reply:
I think this is super clever and am happy to go with it. If we had a bunch more cases of using desugar->sugar, we'd maybe need to step back and take a better look. But given we only needed to overcome a small number of calls into zeroconf, this is ideal IMO.

I do think this would be surprising for a random code reader if they saw this. I'd add a comment -- maybe referencing the ticket for additional context. Just so they can understand that the motivation is to massage the type system and nothing more.

Reply:
I agree with @dgottlieb - I'm fine with the current approach if we have a comment explaining why this is necessary.